extensions/objects/templates/filepath-editor.component.spec.ts

```diff
@@ -0,0 +1,2312 @@
+        'IwIDUyOCAwIDQgLTQxIDg5IC05MCAxOTAKLTE0MyAyOTEgLTI0MiA1MjYgLTMwNS' +
+        'A3MjQgLTE0IDQ3IC01NiAyMDQgLTkxIDM1MCAtNTkgMjQzIC0xMjkgNDkwIC0xMz' +
+        'kgNDkwCi0yIDAgLTYxIC03MyAtMTMyIC0xNjJ6Ii8+CjxwYXRoIGQ9Ik0yNjc5ID' +
+        'QzODMgYy0zMDkgLTMyNCAtNTQwIC03NDEgLTY0OSAtMTE3MiAtNTkgLTIzNCAtOD' +
+        'EgLTQxNCAtODEKLTY3MSAwIC0yMzkgMTYgLTM5MSA2NyAtNjIwIDI4IC0xMjQgMT' +
+        'IxIC00MjAgMTMzIC00MjAgNCAwIDE3IDM5IDMwIDg4IDE0NQo1NjIgMzE2IDkzOS' +
+        'A0NjUgMTAyNyA1OCAzNCAxMjQgMzQgMTgyIDAgMjQxIC0xNDEgNDkyIC05MTYgNz' +
+        'M0IC0yMjY1IDI4IC0xNTcKNTMgLTI5NiA1NyAtMzEwIDYgLTI4IDMgLTQzIDczID' +
+        'M0NSAyMTUgMTE5MiA0MzcgMTkyNSA2NTQgMjE2MiA2NiA3MiAxMDQgOTMKMTY2ID' +
+        'kzIDE3NSAwIDM1MiAtMzE5IDUyNiAtOTQ4IDU5IC0yMTcgNTMgLTIwMSA2NiAtMT' +
+        'c5IDE4IDMzIDc4IDIyMCAxMDggMzM3CjExOCA0NjIgMTE2IDk1MiAtNSAxMzk1IC' +
+        '0xMTUgNDE4IC0zMzkgODIxIC02MjAgMTExNCBsLTk3IDEwMSAtODY4IDAgLTg2Ny' +
+        'AwCi03NCAtNzd6Ii8+CjwvZz4KPC9zdmc+Cg==',
+        originalWidth: 968,
+        originalHeight: 1707,
+        savedImageFilename: 'saved_file_name.png',
+        savedImageUrl: 'assets/images'
+      },
+      crop: false
+    };
+  });
+
+  it('should initialize component correctly', () => {
+    component.ngOnInit();
+
+    expect(component.CROP_CURSORS[component.MOUSE_TOP_LEFT])
+      .toBe('nwse-resize');
+    expect(component.CROP_CURSORS[component.MOUSE_TOP])
+      .toBe('ns-resize');
+    expect(component.CROP_CURSORS[component.MOUSE_TOP_RIGHT])
+      .toBe('nesw-resize');
+    expect(component.CROP_CURSORS[component.MOUSE_RIGHT])
+      .toBe('ew-resize');
+    expect(component.CROP_CURSORS[component.MOUSE_BOTTOM_RIGHT])
+      .toBe('nwse-resize');
+    expect(component.CROP_CURSORS[component.MOUSE_BOTTOM])
+      .toBe('ns-resize');
+    expect(component.CROP_CURSORS[component.MOUSE_BOTTOM_LEFT])
+      .toBe('nesw-resize');
+    expect(component.CROP_CURSORS[component.MOUSE_LEFT])
+      .toBe('ew-resize');
+    expect(component.CROP_CURSORS[component.MOUSE_INSIDE])
+      .toBe('move');
+    expect(component.data).toEqual({
+      mode: component.MODE_EMPTY, metadata: {}, crop: true });
+    expect(component.imageResizeRatio).toBe(1);
+    expect(component.cropArea).toEqual({ x1: 0, y1: 0, x2: 0, y2: 0 });
+    expect(component.mousePositionWithinCropArea).toBe(null);
+    expect(component.mouseLastKnownCoordinates).toEqual({ x: 0, y: 0 });
+    expect(component.lastMouseDownEventCoordinates).toEqual({ x: 0, y: 0 });
+    expect(component.userIsDraggingCropArea).toBe(false);
+    expect(component.userIsResizingCropArea).toBe(false);
+    expect(component.cropAreaResizeDirection).toBe(null);
+    expect(component.invalidTagsAndAttributes).toEqual({tags: [], attrs: []});
+    expect(component.processedImageIsTooLarge).toBe(false);
+    expect(component.entityId).toBe('2');
+    expect(component.entityType).toBe('question');
+  });
+
+  it('should change on mouseup event', () => {
```

Comment:
Maybe re word in this way - `it should change X on.....`

Reply:
Done